### ISSUE TITLE
Editing files with names ending in `/` may not trigger autocommands

### ISSUE THREAD

**Issue by seandewar** [nvim-telescope/telescope.nvim#742]
### Description
For example, on Windows (and maybe other platforms), it's possible for directories in oldfiles to sometimes end in `/`:

![image](https://user-images.githubusercontent.com/6256228/114171491-43cc3800-992c-11eb-9743-ebe8225689bb.png)

Editing such directories via Telescope's oldfiles picker will not trigger events such as `BufEnter`.
This, for example, makes netrw fail to show the directory listing:

![image](https://user-images.githubusercontent.com/6256228/114171654-80982f00-992c-11eb-9e55-e454186500a7.png)

However, `:edit`ing these directories via the command-line does not produce this behaviour:

![image](https://user-images.githubusercontent.com/6256228/114171812-b6d5ae80-992c-11eb-8752-b8944b2fd5be.png)

This is due to some quirky behaviour in Nvim (inherited from Vim) [here](https://github.com/neovim/neovim/blob/cf6c23fb0f1e67484103da9fc583abfbc6e6db91/src/nvim/fileio.c#L366):

```c
// If the name ends in a path separator, we can't open it.  Check here,
// because reading the file may actually work, but then creating the
// swap file may destroy it!  Reported on MS-DOS and Win 95.
if (after_pathsep((const char *)fname, (const char *)(fname + namelen))) {
  filemess(curbuf, fname, (char_u *)_(msg_is_a_directory), 0);
  msg_end();
  msg_scroll = msg_save;
  return FAIL;
}
```

`:edit` deliberately errors if the file name given ends in a path separator.

This can cause issues if `:edit` is invoked in contexts that abort on errors, such as `vim.cmd()`, `nvim_command()`, `nvim_exec()`, `:try`, etc., as subsequent events like `BufEnter` won't be fired.

Telescope uses `vim.cmd()` [here](https://github.com/nvim-telescope/telescope.nvim/blob/d0cf646f65746415294f570ec643ffd0101ca3ab/lua/telescope/actions/set.lua#L116):

```lua
vim.cmd(string.format("%s %s", command, filename))
```

To execute `:edit` as `command` from [here](https://github.com/nvim-telescope/telescope.nvim/blob/d0cf646f65746415294f570ec643ffd0101ca3ab/lua/telescope/actions/init.lua#L178):

```lua
function actions.file_edit(prompt_bufnr)
  return action_set.edit(prompt_bufnr, "edit")
end
```

_(This `:tabedit` [here](https://github.com/nvim-telescope/telescope.nvim/blob/d0cf646f65746415294f570ec643ffd0101ca3ab/lua/telescope/actions/init.lua#L190) may also be affected by this issue)_

It generally makes sense to abort on errors, but I'd expect Telescope to match the behavour of `:edit`ing via the command-line.

One way this can be fixed is to use `:silent! edit` instead, though this will also suppress normal and error messages (`set shortmess+=F` is an Nvim default, which suppresses normal messages from `:edit` anyway).

Another is maybe to use `nvim_exec()` and wrap the command in a `:function!` (without the `abort` argument)?

I can see an argument for filing this as an upstream issue, so let me know if that's preferred. :smile:

**Expected Behavior**
Editing files ending in `/` via Telescope should trigger events such as `BufEnter`.

**Actual Behavior**
Editing files ending in `/` via Telescope doesn't trigger such events.

### Details

<details><summary>Reproduce</summary>

1. nvim -nu test.vim
2. `:Telescope oldfiles`
3. Select a directory entry ending in `/`.
4. Observe the empty buffer; we should see a netrw directory listing instead.
</details>

<details><summary>Environment</summary>

- nvim --version output:
```
NVIM v0.5.0-dev+1216-g8e496b9df
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```
- Operating system: Windows 10 Build 19042
- Telescope commit: `64e59060b1750d0c86761693b6847c3db07afcd2`

</details>

<details><summary>Configuration</summary>
<p>

```viml
set nocompatible hidden laststatus=2

if !filereadable('/tmp/plug.vim')
  silent !curl --insecure -fLo /tmp/plug.vim
    \ https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
endif

source /tmp/plug.vim
call plug#begin('/tmp/plugged')
Plug 'nvim-lua/popup.nvim'
Plug 'nvim-lua/plenary.nvim'
Plug 'nvim-telescope/telescope.nvim'
call plug#end()

autocmd VimEnter * PlugClean! | PlugUpdate --sync | close
lua << EOF

require('telescope').setup{
  defaults = {
    vimgrep_arguments = {
      'rg',
      '--color=never',
      '--no-heading',
      '--with-filename',
      '--line-number',
      '--column',
      '--smart-case'
    },
    prompt_position = "bottom",
    prompt_prefix = ">",
    selection_strategy = "reset",
    sorting_strategy = "descending",
    layout_strategy = "horizontal",
    layout_defaults = {},
    file_ignore_patterns = {},
    shorten_path = true,
    winblend = 0,
    width = 0.75,
    preview_cutoff = 120,
    results_height = 1,
    results_width = 0.8,
    border = {},
    borderchars = { '─', '│', '─', '│', '╭', '╮', '╯', '╰'},
    color_devicons = true,
    use_less = true,
  }
}
EOF
```

**Comment by Conni2461** [comment]
Sorry for the really late response. I am currently trying to reproduce this issue and can't on the latest neovim release 0.7

Is this still an issue for you?

**Comment by seandewar** [comment]
Thanks for taking a look!
This should be fixed upstream now since v8.2.3453 was ported.
